react/src/components/atoms/forms/InputTextFuzzy/index.js

```diff
@@ -0,0 +1,149 @@
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: this.props.selected || '',
+      suggestions: []
+    };
+    const fuseOptions = this.props.fuseOptions;
+    fuseOptions.keys = this.props.keys;
+    this.fuse = new Fuse(this.props.options, fuseOptions);
+  }
+  handleChange = (e) => {
+    const suggestions = this.fuse.search(e.target.value);
+    this.setState({
+      value: e.target.value,
+      suggestions
+    });
+    if (typeof this.props.onChange === 'function') {
+      this.props.onChange(e);
+    }
+  };
+  renderItem = (suggestion) => {
+    return (
+      <span className="suggestion-content">
+          <span className="name">
+            {
+              suggestion.matches.map((match) => {
+                if(this.props.keys.indexOf(match.key) > -1) {
+                  // Add one to each range to get a proper highlight match.
+                  const ranges = match.indices.map((range) => {
+                    return [
+                      range[0],
+                      range[1] === range[0] ? range[1] : range[1] + 1
+                    ]
+                  });
+                  const parts = parse(match.value, ranges);
+                  return parts.filter(part => {
+                    return part.text.length >= 1;
+                  }).map((part, index) => {
+                    const className = part.highlight ? 'highlight' : null;
+                    const key = `${match.key}.suggestion_${index}`;
+                    return(
+                      <span className={className} key={key}>{part.text}</span>
+                    );
+                  });
+                }
+              })
+            }
+          </span>
+        </span>
+    );
+  };
+  renderItemsContainer = ({ children, containerProps }) => {
+    return(<div className="ma__input-fuzzy" {...containerProps}>{children}</div>)
+  };
+  render() {
+    const autoProps = {
+      items: this.state.suggestions,
+      renderItemsContainer: this.renderItemsContainer,
+      renderItem: this.renderItem,
+      renderItemData: { value: this.state.value },
+      inputProps: {
+        placeholder: this.props.placeholder,
+        onChange: this.handleChange,
+        value: this.state.value,
+        disabled: this.props.disabled
+      },
+      id: this.props.id
+    };
+    autoProps.itemProps = (props) => {
+      const { itemIndex } = props;
+      const suggestion = this.state.suggestions[itemIndex];
+      return {
+        'data-item-index': itemIndex,
+        onMouseDown: event => {
+          this.setState({
+            value: suggestion.item.text,
+            suggestions: []
+          });
+          if (typeof this.props.onSuggestionClick === 'function') {
+            // Suggestion is an object that can contain info on score, matches, etc.
+            this.props.onSuggestionClick({event, suggestion});
+          }
+        }
+      }
+    };
+    const inputTextTypeAheadClasses = classNames({
```

Comment:
if you could update these to be the new input's classname, and then add ma_input-fuzzy to the imported scss that would be ideal...